local-test-suite-example/README.md

```diff
@@ -0,0 +1,157 @@
+
+## Creating the test suite
+
+The scripts included in this example relies on a directory structure where each test in the test suite is in a separate directory named based on a `test-suite-name`-`suffix` pattern. This example uses:
+
+```
+.
+├── spring-cloud-serverless-k8s
+└── spring-cloud-serverless-tap
+```
+
+where `spring-cloud-serverless` is the test suite name and each test has a descriptive suffix.
+
+Each test contains two files:
+
+```
+.
+└── spring-cloud-serverless-tap
+    ├── assertions.sh
+    └── options.json
+```
+
+Where `options.json` file is used to specify the configurable options for generating the accelerator and `assertions.sh` is a script the will be run to check the generated content.
+
+The `options.json` file can be created using the [Tanzu App Accelerator VS Code extension](https://marketplace.visualstudio.com/items?itemName=vmware.tanzu-app-accelerator). There is an "Export Options" button for the "Review and Generate" step. You can use this to export the option values selected in the "Configure Accelerator" step.
+
+Here is an example of the `options.json` file (reformatted for readability):
+
+```json
+{
+  "projectName" : "spring-cloud-serverless",
+  "javaVersion" : "17",
+  "deploymentType" : "workload",
+  "includeBuildToolWrapper" : true
+}
+```
+
+The `assertions.sh` script can then be used to verify the generated content. Here is an example based on the above options:
+
+```sh
+#!/bin/bash
+set -euxo pipefail
+source ../../.github/tests/testFunctions.sh
+
+pushd $1
+
+# Verify TAP files
+assertFileExists ./src/main/resources/application.properties
+assertFileExists ./Tiltfile
+assertFileExists ./.tanzuignore
+assertFileExists ./config/workload.yaml
+assertFileContains ./config/workload.yaml 'value: "17"'
+
+# Check that Kubernetes files don't exist
+assertFileDoesntExist ./kubernetes/deployment.yaml
+assertFileDoesntExist ./kubernetes/service.yaml
+
+# Verify the pom.xml file
+assertFileExists ./pom.xml
+assertFileContains ./pom.xml '<java.version>17</java.version>'
+assertPomHasProjectCoordinates ./pom.xml 'com.example' 'spring-cloud-serverless'
+assertPomHasDependency ./pom.xml 'org.springframework.cloud' 'spring-cloud-function-web'
+
+./mvnw --quiet --no-transfer-progress verify
+
+popd
+```
+
+A few things to mention here:
+
+1. We source a `../../.github/tests/testFunctions.sh` file available from the samples [repo](https://github.com/vmware-tanzu/application-accelerator-samples/blob/main/.github/tests/testFunctions.sh).
+1. The script expects to get the directory for the test as first argument so we can switch to that directory
+1. We can provide as many asserts that we think is reasonable to test the accelerator generated content
+1. We can also provide a build command to verify the project can build and run tests
+
+## Running the test suite
+
+This example has three scripts that are used to drive the test suite runs.
+
+1. `run-test-suite.sh` is the top level script that orchestrates the run
```

Comment:
It can be inferred, but you might want to make it explicit at the bottom of these steps that the user should run`./run-test-suite.sh` to execute the tests.